### ISSUE TITLE
Bluetooth headset not working:  Sap driver initialization failed.

### ISSUE THREAD

**Issue by turboMaCk** [NixOS/nixpkgs#103161]
**Describe the bug**
I'm expriencing regression in bluetooth support for pulseaudio (and bluetooth in general).

From log (`journalctl -eu bluetoot`):

```
Nov 08 20:59:11 nixos-mainframe systemd[1]: Starting Bluetooth service...
Nov 08 20:59:11 nixos-mainframe bluetoothd[1157]: Bluetooth daemon 5.54
Nov 08 20:59:11 nixos-mainframe bluetoothd[1157]: Starting SDP server
Nov 08 20:59:11 nixos-mainframe systemd[1]: Started Bluetooth service.
Nov 08 20:59:11 nixos-mainframe bluetoothd[1157]: Bluetooth management interface 1.14 initialized
Nov 08 20:59:11 nixos-mainframe bluetoothd[1157]: Sap driver initialization failed.
Nov 08 20:59:11 nixos-mainframe bluetoothd[1157]: sap-server: Operation not permitted (1)
Nov 08 20:59:19 nixos-mainframe bluetoothd[1157]: Endpoint registered: sender=:1.56 path=/MediaEndpoint/A2DPSource/VENDOR/LDAC
Nov 08 20:59:19 nixos-mainframe bluetoothd[1157]: Endpoint registered: sender=:1.56 path=/MediaEndpoint/A2DPSource/VENDOR/APTXHD
Nov 08 20:59:19 nixos-mainframe bluetoothd[1157]: Endpoint registered: sender=:1.56 path=/MediaEndpoint/A2DPSource/VENDOR/APTX
Nov 08 20:59:19 nixos-mainframe bluetoothd[1157]: Endpoint registered: sender=:1.56 path=/MediaEndpoint/A2DPSource/AAC
Nov 08 20:59:19 nixos-mainframe bluetoothd[1157]: Endpoint registered: sender=:1.56 path=/MediaEndpoint/A2DPSource/SBC
Nov 08 20:59:19 nixos-mainframe bluetoothd[1157]: Endpoint registered: sender=:1.56 path=/MediaEndpoint/A2DPSink/VENDOR/APTXHD
Nov 08 20:59:19 nixos-mainframe bluetoothd[1157]: Endpoint registered: sender=:1.56 path=/MediaEndpoint/A2DPSink/VENDOR/APTX
Nov 08 20:59:19 nixos-mainframe bluetoothd[1157]: Endpoint registered: sender=:1.56 path=/MediaEndpoint/A2DPSink/AAC
Nov 08 20:59:19 nixos-mainframe bluetoothd[1157]: Endpoint registered: sender=:1.56 path=/MediaEndpoint/A2DPSink/SBC
Nov 08 20:59:33 nixos-mainframe bluetoothd[1157]: 00:1B:66:C0:F6:98: error updating services: Host is down (112)
```

I'm also experiencing regression with connectivity of XBox One Wireless Controller which requires disabling of ertm (`echo 'Y' >| /sys/module/bluetooth/parameters/disable_ertm`) normally but I'm enable to connect it. I haven't tried to connect any other bluetooth device so all I tried don't work at the moment for me.

**To Reproduce**
1. with configuration containing
```
hardware = {
    pulseaudio = {
      # Sound config
      enable = true;
      package = pkgs.pulseaudioFull;
      extraModules = [ pkgs.pulseaudio-modules-bt ];
      support32Bit = true;
      extraConfig = "
        load-module module-switch-on-connect
      ";
    };

    # Bluetooth
    bluetooth = {
      enable = true;
    };
  };
};
```
2. try to connect connect bluetooth headset

**Expected behavior**
Devices should be connected and working.

**Additional context**

**Notify maintainers**
cc @rnhmjoj who worked on Bluetooth recently.

**Metadata**
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```
 - system: `"x86_64-linux"`
 - host os: `Linux 5.4.75, NixOS, 21.03pre250878.a52e974cff8 (Okapi)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.3.8`
 - channels(root): `"nixos-21.03pre250878.a52e974cff8, nixos-hardware"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

Maintainer information:
```yaml
# a list of nixpkgs attributes affected by the problem
attribute:
# a list of nixos modules affected by the problem
module:
```

**Comment by rnhmjoj** [comment]
> cc @rnhmjoj who worked on Bluetooth recently.

Sorry, I don't think I can help you much: I did work on the bluetooth module but my bluetooth stack knowledge is near 0. 

> I'm expriencing regression in bluetooth support for pulseaudio (and bluetooth in general).

It's not clear to me what the problem is, exactly. Is the bluetooth audio output not working? Is hhe bluethooth service failing or can't pair the devices?

From the log you posted I don't see anything wrong:

```
Nov 08 20:59:11 nixos-mainframe bluetoothd[1157]: Sap driver initialization failed.
```
From a quick search, it appears to be a simple warning and not a fatal error. Unless you want to connect a SIM card remotely, it should be of no concern.

```
Nov 08 20:59:33 nixos-mainframe bluetoothd[1157]: 00:1B:66:C0:F6:98: error updating services: Host is down (112)
```
This is an actual error, but only tells a device disconnected.

You're saying this is a regression but did not mention updating your system/changing configuration. When did this start? In any case, bluez was last updated in March, so it can be ruled out if this only started recently. On the other hand, pulseaudio-ctl has been updated a few days ago.

**Comment by mfsch** [comment]
I also just ran into issues with my bluetooth mouse (MX Master) which wouldn’t work anymore after some recent updates after working fine for months. I don’t know if those are related, but I’m posting it here just in case, including how I got it to work again. Apart from the `sap-server` error (which does indeed appear to be unrelated in my case), I also had an error `input-hog profile accept failed for D5:XX:XX:XX:XX:XX` in the logs. What helped in my case was to run `sudo modprobe uhid` as recommended [here](https://forums.gentoo.org/viewtopic-t-1073060-start-0.html) and [here](https://github.com/clearlinux/distribution/issues/956), and the permanent fix was adding `"uhid"` to `boot.kernelModules` in my NixOS config. Perhaps this module used to be loaded automatically? I think in my case the issue coincided with the kernel update from 5.9.10 to 5.9.11.
